docs/user_guide/mitigation.rst

```diff
@@ -80,11 +81,408 @@ Reductions
+
+    \P[h(X) = 1 \given A = a, Y = 0] = \P[h(X) = 1 \given Y = 0] \; \forall a
+
+Note that this formulation of True Positive Rate Parity ignores what happens
+to samples with :math:`Y=0`, and the False Positive Rate Parity ignores
+:math:`Y=1`.
+Refer to :ref:`equalized_odds` for a constraint that extends this by
+considering all samples, i.e., the samples with :math:`Y=0` as well as the
+ones with :math:`Y=1`.
+
+In practice this can be used in a difference-based version as follows:
+
+.. doctest:: mitigation
+
+    >>> from fairlearn.reductions import TruePositiveRateParity
+    >>> from fairlearn.metrics import true_positive_rate_group_summary
+    >>> import numpy as np
+    >>> tprp = TruePositiveRateParity(difference_bound=0.01)
+    >>> X                  = np.array([[0], [1], [2], [3], [4], [5], [6], [7], [8], [9]])
+    >>> y_true             = np.array([ 1 ,  1 ,  1 ,  1 ,  1,   1 ,  1 ,  0 ,  0 ,  0 ])
+    >>> y_pred             = np.array([ 1 ,  1 ,  1 ,  1 ,  0,   0 ,  0 ,  1 ,  0 ,  0 ])
+    >>> sensitive_features = np.array(["a", "b", "a", "a", "b", "a", "b", "b", "a", "b"])
+    >>> true_positive_rate_group_summary(y_true, y_pred, sensitive_features=sensitive_features)
+    {'overall': 0.5714285714285714, 'by_group': {'a': 0.75, 'b': 0.3333333333333333}}
+    >>> tprp.load_data(X, y_true, sensitive_features=sensitive_features)
+    >>> tprp.gamma(lambda X: y_pred)
+    sign  event    group_id
+    +     label=1  a           0.178571
+                   b          -0.238095
+    -     label=1  a          -0.178571
+                   b           0.238095
+    dtype: float64
+
+.. note::
+
+    In mitigation examples we never need to call :code:`load_data` or
+    :code:`gamma` since the reductions techniques all automatically take care
+    of it. The example merely helps in illustrating the deviation from the
+    corresponding inequalities in :ref:`constraints_binary_classification`.
+
+Similarly to other constraints a ratio-based version is also available:
+
+.. doctest:: mitigation
+
+    >>> tprp = TruePositiveRateParity(ratio_bound=0.9, ratio_bound_slack=0.01)
+    >>> tprp.load_data(X, y_true, sensitive_features=sensitive_features)
+    >>> tprp.gamma(lambda X: y_pred)
+    sign  event    group_id
+    +     label=1  a           0.103571
+                   b          -0.271429
+    -     label=1  a          -0.235714
+                   b           0.180952
+    dtype: float64
+
+.. _equalized_odds:
+    
+Equalized Odds
+^^^^^^^^^^^^^^
+
+A classifier :math:`h(X)` satisfies *Equalized Odds* if it satisfies both
+*True Positive Rate Parity* and *False Positive Rate Parity*, i.e.,
+
+.. math::
+
+    \P[h(X) = y' \given A = a, Y = y] = \P[h(X) = y' \given Y = y] \; \forall a, y, y'
+
+From a usage standpoint this works just like :ref:`true_positive_rate_parity`.
+The constraint violations with :code:`label=1` match exactly the ones from
+True Positive Rate Parity above, while the ones with :code:`label=0` represent
```

Comment:
Need to point out that the reader needs to look at the _difference_ code boxes, not the _ratio_ ones.

Reply:
This text doesn't quite exist anymore after Miro's revisions